internal/systemd/systemd_test.go

```diff
@@ -0,0 +1,95 @@
+package systemd_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/systemd"
+	"github.com/ubuntu/adsys/internal/testutils"
+)
+
+var ctx = context.Background()
+
+func TestStartUnit(t *testing.T) {
+	t.Parallel()
+
+	bus := testutils.NewDbusConn(t)
+
+	systemdCaller, err := systemd.NewWithConn(bus)
+	require.NoError(t, err, "Setup: failed to create systemd caller")
+
+	err = systemdCaller.StartUnit(ctx, "service-1.service")
+	require.NoError(t, err, "StartUnit shouldn't have failed but it did")
+
+	err = systemdCaller.StartUnit(ctx, "not-a-service.service")
+	require.Error(t, err, "StartUnit should have failed for nonexistent service")
+
+	err = systemdCaller.StartUnit(ctx, "fail-to-start-stop.service")
```

Comment:
small like a good candidate for table testing to test all of those use cases separately :) (same on other tests)